plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/transport/simulate/SimulatedItemResponse.java

```diff
@@ -0,0 +1,169 @@
+
+    private Data data;
+    private List<ProcessorResult> processorResultList;
+    private Throwable failure;
+
+    public SimulatedItemResponse() {
+
+    }
+
+    public SimulatedItemResponse(Data data) {
+        this.data = data;
+    }
+
+    public SimulatedItemResponse(List<ProcessorResult> processorResultList) {
+        this.processorResultList = processorResultList;
+    }
+
+    public SimulatedItemResponse(Throwable failure) {
+        this.failure = failure;
+    }
+
+    public boolean isFailed() {
+        if (failure != null) {
+            return true;
+        } else if (processorResultList != null) {
+            for (ProcessorResult result : processorResultList) {
+                if (result.isFailed()) {
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    public boolean isVerbose() {
+        return this.processorResultList != null;
+    }
+
+    public Data getData() {
+        return data;
+    }
+
+    public List<ProcessorResult> getProcessorResultList() {
+        return processorResultList;
+    }
+
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        boolean isFailed = in.readBoolean();
+        boolean isVerbose = in.readBoolean();
+        if (isFailed) {
+            this.failure = in.readThrowable();
+        } else if (isVerbose) {
+            int size = in.readVInt();
+            processorResultList = new ArrayList<>();
+            for (int i = 0; i < size; i++) {
+                ProcessorResult processorResult = new ProcessorResult();
+                processorResult.readFrom(in);
+                processorResultList.add(processorResult);
+            }
+        } else {
+            String index = in.readString();
+            String type = in.readString();
+            String id = in.readString();
+            Map<String, Object> doc = in.readMap();
+            this.data = new Data(index, type, id, doc);
+        }
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeBoolean(isFailed());
+        out.writeBoolean(isVerbose());
+
+        if (failure != null) {
+            out.writeThrowable(failure);
+        } else if (isVerbose()) {
+            out.writeVInt(processorResultList.size());
+            for (ProcessorResult p : processorResultList) {
+                p.writeTo(out);
+            }
+        } else {
+            out.writeString(data.getIndex());
+            out.writeString(data.getType());
+            out.writeString(data.getId());
+            out.writeMap(data.getDocument());
+        }
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(Fields.ERROR, isFailed());
+        if (failure != null) {
+            builder.field(Fields.ERROR_MESSAGE, ExceptionsHelper.detailedMessage(failure));
```

Comment:
should we use ElasticsearchException#renderThrowable instead?Seems like it would make things more consistent on the REST layer. Looking at the MultiGetResponse#toXContent I think we could drop the initial boolean flag as well?